File: web/models/post.js
import mongoose from "mongoose";

const postSchema = new mongoose.Schema({
    text: {
        type: String,
        required: [true, "Post is required"],
    },
    author: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "User",
        required: true,
    },
    createdAt: {
        type: Date,
        default: Date.now,
    },
    parentId: {
        type: String,
    },
    likeCount: {
        type: Number,
        default: 0,
    },
    children: [
        {
            type: mongoose.Schema.Types.ObjectId,
            ref: "Post",
        },
    ],
});

const Post = mongoose.models.Post || mongoose.model("Post", postSchema);

export default Post;

File: android/app/src/main/java/com/mr_17/mednex/ui/community/models/Post.kt
package com.mr_17.mednex.ui.community.models

import com.google.gson.annotations.SerializedName

data class Post(
    @SerializedName("author")
    val authorId: String,
    @SerializedName("children")
    val childrenIdList: List<String>,
    @SerializedName("createdAt")
    val createdAt: String,
    @SerializedName("_id")
    val id: String,
    @SerializedName("parentId")
    val parentId: String,
    @SerializedName("text")
    val text: String,
    @SerializedName("__v")
    val v: Int
)